enterprise/server/remote_execution/containers/firecracker/firecracker.go

```diff
@@ -329,6 +329,10 @@ func (p *Provider) New(ctx context.Context, props *platform.Properties, task *re
 			InitDockerd:       props.InitDockerd,
 			EnableDockerdTcp:  props.EnableDockerdTCP,
 		}
+	} else if *snaploader.EnableLocalSnapshotSharing {
+		// When local snapshot sharing is enabled, reject old-style persisted
+		// snapshots, since these aren't shareable (i.e. COW-formatted).
+		return nil, status.UnavailableError("ignoring persisted snapshot; this functionality has been replaced by local snapshot sharing")
```

Comment:
Should this return nil? Or should it just use the default vmConfig like when savedState == nil

Reply:
These VMs are eagerly added to the pool on executor startup and if we return an error then we just skip adding it. So I think returning an error makes sense here